src/app/core/services/auth.service.ts

```diff
@@ -133,6 +135,37 @@ export class AuthService {
     this.logger.info(`AuthService: Redirecting for Github authentication`);
   }
 
+  /**
+   * Handles the clean up required after authentication and setting up of repository is completed.
+   */
+  handleSetRepoSuccess() {
+    this.setTitleWithPhaseDetail();
+    this.router.navigateByUrl(Phase.issuesViewer);
+  }
+
+  /**
+   * Setup repository after authentication.
+   */
+  setRepo(): void {
+    this.phaseService.initializeCurrentRepository();
+    const currentRepo = this.phaseService.currentRepo;
+    this.githubService.isRepositoryPresent(currentRepo.owner, currentRepo.name)
+      .pipe(
+        mergeMap((isValidRepository) => {
+          if (!isValidRepository) {
+            return new Observable();
+          }
+          return this.githubEventService.setLatestChangeEvent();
+        })
+      )
+      .subscribe(
+        () => {
+          this.handleSetRepoSuccess();
+        }
```

Comment:
```suggestion
          if (!isValidRepository) {
            throw new Error("Repo is invalid!");
          }
          return this.githubEventService.setLatestChangeEvent();
        })
      )
      .subscribe(
        () => this.handleSetRepoSuccess(),
        (error) => {
          // some kind of error handling
          this.logger.error(`AuthService: Error while setting repository: ${error}`);
        }
```

Might be more reasonable to rephrase this since this subscription will be pointless as it will always run `this.handleSetRepoSuccess()` regardless of whether or not the repo is valid.

Reply:
I played around with the implementation of `setRepo` and decided to shift the check for `isValidRepository` into `PhaseService::initializeCurrentRepository`, so that no repo is set if the repo provided is invalid. `handleSetRepoSuccess` is also now only called if a repo is set in `initializeCurrentRepository`. I've also made some changes in `session-selection.component.ts` so users can try to set a valid repository if they previously provided an invalid one.

Reply:
> There seems to be 'other forces' at work here as the application is still navigating to issueViewer before `handleSetRepoSuccess()` is not called. (not sure why this is the case)

Also, if you happen to mean the phase descriptor in the header component changing, then I think its because it displays the current phase once the user is authenticated, which by default is the issue dashboard. I'm not sure if i should change it.

Reply:
Ah I see, I think it's fine for now.